applications/vanilla/models/class.discussionmodel.php

```diff
@@ -3066,7 +3066,9 @@ public function deleteID($discussionID, $options = []) {
         $this->SQL->delete('Discussion', ['DiscussionID' => $discussionID]);
 
         $this->SQL->delete('UserDiscussion', ['DiscussionID' => $discussionID]);
-        $this->updateDiscussionCount($categoryID);
+        /** @var Vanilla\Scheduler\SchedulerInterface $scheduler */
+        $scheduler = Gdn::getContainer()->get(Vanilla\Scheduler\SchedulerInterface::class);
+        $scheduler->addJob(Vanilla\Library\Jobs\UpdateDiscussionCount::class, ['categoryID' => $categoryID]);
```

Comment:
Shouldn't this go inside of the existing `updateDiscussionCount` method?

Currently there may still be places that are updating it sychronously.

Reply:
Makes sense. There are multiple places where we call this method. I'll update it.